pydeequ/configs.py

```diff
@@ -1,44 +1,36 @@
 # -*- coding: utf-8 -*-
-import logging
-import os
+from functools import lru_cache
+import subprocess
+import re
 
-logger = logging.getLogger("logger")
-configs = {
-    "deequ_maven_coord": "com.amazon.deequ:deequ:1.2.2-spark-3.0",
-    "deequ_maven_coord_spark3": "com.amazon.deequ:deequ:1.2.2-spark-3.0",
-    # "deequ_maven_coord_spark2_4": "com.amazon.deequ:deequ:1.2.2-spark-2.4", # 1.2.2 is broken, rolling back to 1.1.0 with scala 11
-    "deequ_maven_coord_spark2_4": "com.amazon.deequ:deequ:1.1.0_spark-2.4-scala-2.11",
-    # "deequ_maven_coord_spark2_2": "com.amazon.deequ:deequ:1.2.2-spark-2.2",
-    "deequ_maven_coord_spark2_2": "com.amazon.deequ:deequ:1.1.0_spark-2.2-scala-2.11",
-    "f2j_maven_coord": "net.sourceforge.f2j:arpack_combined_all",
+SPARK_TO_DEEQU_COORD_MAPPING = {
+    "3.2": "com.amazon.deequ:deequ:2.0.1-spark-3.2",
+    "3.1": "com.amazon.deequ:deequ:2.0.0-spark-3.1",
+    "3.0": "com.amazon.deequ:deequ:1.2.2-spark-3.0",
+    "2.4": "com.amazon.deequ:deequ:1.1.0_spark-2.4-scala-2.11",
 }
 
 
-def _get_spark_version():
-    # TODO - Change this later [Use Spark API's instead of env var]
-    spark_version: str = os.getenv("SPARK_VERSION")
+@lru_cache(maxsize=None)
+def _get_spark_version() -> str:
+    # Get version from a subprocess so we don't mess up with existing SparkContexts.
+    command = [
+        "python",
+        "-c",
+        "from pyspark import SparkContext; print(SparkContext.getOrCreate()._jsc.version())",
+    ]
+    output = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    spark_version = output.stdout.decode().split("\n")[-2]
     return spark_version
 
 
-def set_deequ_maven_config():
+def _get_deequ_maven_config():
     spark_version = _get_spark_version()
-    if spark_version is None:
-        logger.error("Please set env variable SPARK_VERSION")
-        logger.info(f"Using deequ: {configs['deequ_maven_coord']}")
-        return configs["deequ_maven_coord"]  # TODO
-    if spark_version[0:3] == "3.0":
-        logger.info("Setting spark-3.0 as default version of deequ")
-        configs["deequ_maven_coord"] = configs["deequ_maven_coord_spark3"]
-    elif spark_version[0:3] == "2.4":
-        logger.info("Setting spark-2.4 as default version of deequ")
-        configs["deequ_maven_coord"] = configs["deequ_maven_coord_spark2_4"]
-    elif spark_version[0:3] == "2.2":
-        logger.info("Setting spark3 as default version of deequ")
-        configs["deequ_maven_coord"] = configs["deequ_maven_coord_spark2_2"]
-    else:
-        logger.error(f"Deequ is still not supported in spark version: {spark_version}")
-        logger.info(f"Using deequ: {configs['deequ_maven_coord']}")
-        return configs["deequ_maven_coord"]  # TODO
+    try:
+        return SPARK_TO_DEEQU_COORD_MAPPING[spark_version[:3]]
+    except KeyError:
+        raise RuntimeError(f"Deequ is still not supported in spark version {spark_version}")
```

Comment:
Nit: let's print out supported versions? 
```suggestion
        raise RuntimeError(f"Found Incompatible Spark version {spark_version}; Use one of the Supported Spark versions for Deequ: {SPARK_TO_DEEQU_COORD_MAPPING.keys()}")
```